src/components/Index/BillContent.svelte

```diff
@@ -15,13 +15,53 @@
 		bills: Pick<Bill, 'id' | 'title' | 'nickname'>[];
 	}
 
+	type MpTerm = {
+		name: string;
+		custom_name?: string;
+		founding_date: string | null;
+		dissolution_date: string | null;
+		term: number | null;
+	};
+
+	let mpterm_display_name = (mp: MpTerm) => {
+		const s = `สส. ชุดที่ ${mp.term}`;
+
+		const toBE2Digits = (year?: number) =>
+			year ? ((year + 543) % 100).toString().padStart(2, '0') : undefined;
+
+		const foundingYear = mp.founding_date
+			? toBE2Digits(new Date(mp.founding_date).getFullYear())
+			: undefined;
+		const dissolutionYear = mp.dissolution_date
+			? toBE2Digits(new Date(mp.dissolution_date).getFullYear())
+			: undefined;
+
+		if (!foundingYear && !dissolutionYear) return s;
+		if (!foundingYear && dissolutionYear) return `${s} (ก่อนปี ${dissolutionYear})`;
+		if (foundingYear && dissolutionYear) return `${s} (ปี ${foundingYear} - ${dissolutionYear})`;
+		if (foundingYear && !dissolutionYear) return `${s} (ปี ${foundingYear} - ปัจจุบัน)`;
+
+		return s;
+	};
+
+	export let billCategories: string[] = [];
+	export let mpTerms: MpTerm[] = [];
+
 	const ALL_CATEGORY_KEY = 'ทุกหมวด';
+	const OTHER_TERM_KEY: MpTerm = {
+		name: 'other_terms',
+		custom_name: 'สส. ชุดอื่น ๆ (ก่อนปี 62)',
+		founding_date: null,
+		dissolution_date: '2019-03-24',
+		term: 0
+	};
 	const MAX_BILL_BY_STATUS = 3;
 	const MAX_ENACTED_BILL = 10;
 
-	export let billCategories: string[] = [];
+	let all_mp_choices = [...mpTerms, OTHER_TERM_KEY];
```

Comment:
`mpTerms` is reactive because it's a props. If you use `let` and mpTerms changes, this variable won't. Use svelte reactive statement every time something is depended on reactive variable.

```svelte
$: allMpChoices = [...mpTerms, OTHER_TERM_KEY];
```

PS. this `$` was removed in Svelte latest version but don't worry we didn't migrate yet.